lib/bom.js

```diff
@@ -368,6 +368,11 @@ declare class CloseEvent extends Event {
     wasClean: boolean;
 }
 
+type MessageEventHandler = (event: MessageEvent) => mixed;
+type CloseEventHandler = (event: CloseEvent) => mixed;
+type MessageEventListener = {handleEvent: MessageEventHandler} | MessageEventHandler;
```

Comment:
I worry that overriding `addEventListener` is going to hit the same perf regression that https://github.com/facebook/flow/pull/2680 hit.